browser-interface/packages/shared/session/sagas.ts

```diff
@@ -321,7 +321,9 @@ async function createAuthIdentity(requestManager: RequestManager, isGuest: boole
 
   const ssoIdentity = await SingleSignOn.getIdentity(address)
 
-  if (!ssoIdentity) {
+  const validIdentity = !ssoIdentity ? false : isSessionExpired({ identity: ssoIdentity })
+
+  if (!ssoIdentity || validIdentity) {
```

Comment:
```suggestion
  if (validIdentity) {
```
This should be enough I believe